src/components/SelectInput.jsx

```diff
@@ -0,0 +1,118 @@
+import React, { useRef, useState } from 'react'
+import PropTypes from 'prop-types'
+import { makeStyles } from '@material-ui/core/styles'
+import ActionMenu, {
+  ActionMenuItem,
+  ActionMenuRadio
+} from 'cozy-ui/transpiled/react/ActionMenu'
+import DropdownText from 'cozy-ui/transpiled/react/DropdownText'
+import InputGroup from 'cozy-ui/transpiled/react/InputGroup'
+import useBreakpoints from 'cozy-ui/transpiled/react/hooks/useBreakpoints'
+
+const useStyles = makeStyles(() => ({
+  mobileInput: {
+    width: '100%',
+    maxWidth: 'none',
+    border: '0',
+    '&:hover, &:focus, &:active': {
+      border: '0'
+    }
+  },
+  desktopInput: {
+    height: '2.5rem',
+    borderRadius: '1.25rem',
+    flex: '0'
+  },
+  mobileButton: {
+    boxSizing: 'border-box',
+    padding: '.625rem .5rem',
+    lineHeight: '1.25rem',
+    whiteSpace: 'nowrap',
+    '& span, & p': {
+      display: 'inline'
+    }
+  },
+  desktopButton: {
+    boxSizing: 'border-box',
+    margin: '-.063rem',
+    padding: '.625rem 1rem',
+    borderRadius: '1.25rem',
+    lineHeight: '1.25rem',
+    whiteSpace: 'nowrap',
+    '& span, & p': {
+      display: 'inline'
+    }
+  }
+}))
+
+const SelectInput = ({ options, name, value, setValue }) => {
+  const { isMobile } = useBreakpoints()
+  const styles = useStyles()
+  const [open, setOpen] = useState(false)
+
+  const handleDropdownTextClick = () => {
+    setOpen(true)
+  }
+
+  const handleActionMenuClose = () => {
+    setOpen(false)
+  }
+
+  const handleActionMenuItemClick = value => {
+    setValue(value)
+  }
+
+  const anchorRef = useRef()
+
+  return (
+    <>
+      <InputGroup
+        className={isMobile ? styles.mobileInput : styles.desktopInput}
+      >
+        <DropdownText
+          className={isMobile ? styles.mobileButton : styles.desktopButton}
+          variant="caption"
+          onClick={handleDropdownTextClick}
+          ref={anchorRef}
+        >
+          {options[value]}
+        </DropdownText>
+        {open && (
+          <ActionMenu
+            className={styles.menu}
+            anchorElRef={anchorRef}
+            autoclose={true}
+            onClose={handleActionMenuClose}
+          >
+            {Object.entries(options).map(([optionValue, optionLabel]) => {
+              return (
+                <ActionMenuItem
+                  key={optionValue}
+                  left={
+                    <ActionMenuRadio
+                      name={name}
+                      value={optionValue}
+                      checked={optionValue === value}
+                    />
+                  }
+                  onClick={() => handleActionMenuItemClick(optionValue)}
+                >
+                  {optionLabel}
+                </ActionMenuItem>
+              )
+            })}
+          </ActionMenu>
+        )}
+      </InputGroup>
+    </>
+  )
+}
+
+SelectInput.propTypes = {
+  options: PropTypes.object.isRequired,
```

Comment:
can we use PropTypes.shape or better, PropTypes.exact here?

Reply:
This is a dictionnary with string values (this would be a `Record<any, string>` in TypeScript).
To narrow it further, i put `PropTypes.objectOf(PropTypes.arrayOf(PropTypes.string))`  instead